### PR TITLE
[Python] Fix `examples/duckdb-python.py`

### LINKED ISSUE #8199: run the duckdb-python.py in example folder occurs 3 errors
### What happens?

# 1
CleanShot 2023-07-10 at 15 39 17
fix by 
CleanShot 2023-07-10 at 15 34 57
# 2
similarly in line 137, i fix it by
CleanShot 2023-07-10 at 15 35 44
# 3
CleanShot 2023-07-10 at 15 37 01
fix by
CleanShot 2023-07-10 at 15 38 14

### To Reproduce

```python
 python3 duckdb-python.py
```

### OS:

mac

### DuckDB Version:

0.8.2.dev1549

### DuckDB Client:

Python

### Full Name:

Xuanwei Zhang

### Affiliation:

Gatech

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #8199 

Addressed the issues, also added another example to the section that focuses on querying Pandas DataFrames, using a replacement scan.

### CODE REVIEW

**Comment by Mause** [comment]
Maybe we can run this as a test via pytest to avoid it breaking in the future?

**Comment by Tishj** [comment]
> Maybe we can run this as a test via pytest to avoid it breaking in the future?

That's a good idea, I'm just not sure how this would be implemented
do you just want to import the file relative to the `pythonpkg` module or are you thinking of bundling it as part of the python package so we can import it that way?

File: tools/pythonpkg/tests/fast/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import subprocess
+import os
+
+class TestExamples(object):
+	def test_examples(self):
+		current_dir = os.path.dirname(os.path.abspath(__file__))
+		relative_path = "../../../../examples/python/duckdb-python.py"
+		absolute_path = os.path.abspath(os.path.join(current_dir, relative_path))
+		result = subprocess.run(['python3', absolute_path])
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/test_examples.py:11]: This will fail on windows - just using `importlib.import_module()` or similiar should suffice

**Comment by Tishj** [comment]
This is complicated by `cibuildwheel`, because it creates a separate environment, so this relative path is not working.

I don't think this is possible unless we move this file into the tests directory, which kind of defeats the point.
Granted I don't see the point of this file anyways since our tests are already provide much better examples, but that's besides the point.

**Pull request closed without merging**